cmd/net.go

```diff
@@ -98,6 +108,10 @@ func mustGetLocalIP6() (ipList set.StringSet) {
 func getHostIP(host string) (ipList set.StringSet, err error) {
 	var ips []net.IP
 
+	// net.LookupIP requires IPv6 not include brackets
+	if strings.HasPrefix(host, "[") && strings.HasSuffix(host, "]") {
+		host = host[1 : len(host)-1]
```

Comment:
we should rely on net.SplitHostPort() to get the host portion appropriately, manually adding this kind of code is not useful.

Reply:
@harshavardhana This code is necessary because if you pass an IPv6 address without a port (i.e. `[::2001:4860:4860:8888]`) to `updateDomainIPs()` then `net.SplitHostPort()` fails with the error 'missing port in address'. As such, the `host` variable is left unmodified and still contains the brackets (which `LookupIP` chokes on).  Therefore, either `updateDomainIPs()` needs to strip the brackets for, or alternatively, `getHostIP()` should be made safe so it can be called as `getHostIP("[::2001:4860:4860:8888]")` or `getHostIP("::2001:4860:4860:8888")`.

I chose to make `getHostIP()`  more tolerant of various input formats rather than adding this specialized logic to `updateDomainIPs()` and thus risking that other places using `getHostIP()` would need similar fixes.

In other words, if you don't include this fix the following would fail:

```
domainIPs : = CreateStringSet("[::2001:4860:4860:8844]"),
updateDomainIPs(domainIPs)
```

The alternative fix would be to use this code:

```
host, port, err := net.SplitHostPort(e)
if err != nil {
	if strings.Contains(err.Error(), "missing port in address") {
		host = e
                # getHostIP does not allow IPv6 address to include brackets
                if IsIPv6(host) {
                  host = StripIPv6Brackets(host)
                }
		port = globalMinioPort
	} else {
		continue
	}
}
```

Reply:
```

Reply:
I don't think I can do this:

```
updateDomainIPs(localIP4)
updateDomainIPs(localIP6)
```

Because the `updateDomainIPs` method replaces the `globalDomainIPs` set rather than augmenting it (which was existing behavior not modified in this PR).

What do you think about allowing `updateDomainIPs` to be passed multiple endpoint sets.

```
updateDomainIPs(localIP4, localIP6)
```

Reply:
The function was meant for that purpose, it is meant to update not set as the function name. It is not updating but replacing which is unfortunate..

Reply:
This updated PR no longer includes anything related to IP6 and it maintains the original behavior in `updateDomainIPs`.  I'll leave the topic of changing the behavior of `updateDomainIPs` for you to decide.